kornia/enhance/jpeg.py

```diff
@@ -0,0 +1,573 @@
+from __future__ import annotations
+
+from typing import Optional
+
+import torch
+
+from kornia.color import rgb_to_ycbcr, ycbcr_to_rgb
+from kornia.core import Device, Dtype, Module, Tensor
+from kornia.core.check import (
+    KORNIA_CHECK,
+    KORNIA_CHECK_IS_TENSOR,
+    KORNIA_CHECK_SHAPE,
+)
+from kornia.geometry import rescale
+
+__all__ = ["jpeg_encode_decode_differentiable", "JPEGEncodeDecodeDifferentiable"]
+
+
+def _get_default_qt_y(device: Device, dtype: Dtype) -> Tensor:
+    """Generates default Quantization table of Y channel."""
+    return torch.tensor(
+        [
+            [16, 11, 10, 16, 24, 40, 51, 61],
+            [12, 12, 14, 19, 26, 58, 60, 55],
+            [14, 13, 16, 24, 40, 57, 69, 56],
+            [14, 17, 22, 29, 51, 87, 80, 62],
+            [18, 22, 37, 56, 68, 109, 103, 77],
+            [24, 35, 55, 64, 81, 104, 113, 92],
+            [49, 64, 78, 87, 103, 121, 120, 101],
+            [72, 92, 95, 98, 112, 100, 103, 99],
+        ],
+        device=device,
+        dtype=dtype,
+    )
+
+
+def _get_default_qt_c(device: Device, dtype: Dtype) -> Tensor:
+    """Generates default Quantization table of C channels."""
+    return torch.tensor(
+        [
+            [17, 18, 24, 47, 99, 99, 99, 99],
+            [18, 21, 26, 66, 99, 99, 99, 99],
+            [24, 26, 56, 99, 99, 99, 99, 99],
+            [47, 66, 99, 99, 99, 99, 99, 99],
+            [99, 99, 99, 99, 99, 99, 99, 99],
+            [99, 99, 99, 99, 99, 99, 99, 99],
+            [99, 99, 99, 99, 99, 99, 99, 99],
+            [99, 99, 99, 99, 99, 99, 99, 99],
+        ],
+        device=device,
+        dtype=dtype,
+    )
+
+
+def _patchify_8x8(input: Tensor) -> Tensor:
```

Comment:
would it be worth using https://kornia.readthedocs.io/en/latest/contrib.html#kornia.contrib.extract_tensor_patches and the `combine_tensor_patches` (instead of `_patchify_8x8` and ``_unpatchify_8x8) here? cc @edgarriba

Reply:
When I import `combine_tensor_patches` using `from kornia.contrib import extract_tensor_patches` I receive a circular import error 🤔

Reply:
Try with the full path `from kornia.contrib.extract_patches import extract_tensor_patches`

Reply:
Unfortunately, loading `extract_patches` directly also leads to a circular import...

```
from kornia.enhance import normalize_min_max
E   ImportError: cannot import name 'normalize_min_max' from partially initialized module 'kornia.enhance' (most likely due to a circular import)
```

Reply:
Just importing kornia and calling `kornia.enhance.extract_tensor_patches` works...